src/preferences.cpp

```diff
@@ -34,6 +34,7 @@
 
 #include <sys/stat.h> // for setting the permissions of the preferences file
 #include <boost/concept_check.hpp>
+#include <unistd.h>
```

Comment:
can you please put #ifndef _WIN32 around this? Otherwise you'll break the build on windows.